Shell/Shell.cpp

```diff
@@ -1373,6 +1373,10 @@ void Shell::cache_path()
         }
     }
 
+    // add shell builtins to the cache
+    for (const auto& builtin_name : builtins())
+        cached_path.append(escape_token(builtin_name));
```

Comment:
I don't think you need the separate function for this, you can just iterate the array with range-for:

```cpp
for (auto* builtin_name : builtin_names)
    ...
```

Reply:
range-based for works with constexpr arrays? neat!